crates/ide-db/src/syntax_helpers/node_ext.rs

```diff
@@ -52,7 +52,10 @@ pub fn preorder_expr(start: &ast::Expr, cb: &mut dyn FnMut(WalkEvent<ast::Expr>)
             }
         };
         if let Some(let_stmt) = node.parent().and_then(ast::LetStmt::cast) {
-            if Some(node.clone()) != let_stmt.initializer().map(|it| it.syntax().clone()) {
+            let node = Some(node.clone());
+            if node != let_stmt.initializer().map(|it| it.syntax().clone())
+                && node != let_stmt.let_else().map(|it| it.syntax().clone())
+            {
```

Comment:
Not sure why the previous implementation was cloning — maybe this would be better? Or is cloning trivially cheap? 
```suggestion
            if let_stmt.initializer().map(|it| it.syntax() != &node).unwrap_or(false)
                && let_stmt.let_else().map(|it| it.syntax() != &node).unwrap_or(false)
            {
```

Reply:
SyntaxNodes are ref counted pointers so they are cheap to clone, that's why